src/Kestrel.Core/Internal/ConnectionHandler.cs

```diff
@@ -24,38 +28,73 @@ public ConnectionHandler(ListenOptions listenOptions, ServiceContext serviceCont
             _application = application;
         }
 
-        public IConnectionContext OnConnection(IConnectionInformation connectionInfo)
+        public void OnConnection(IFeatureCollection features)
         {
-            var inputPipe = connectionInfo.PipeFactory.Create(GetInputPipeOptions(connectionInfo.InputWriterScheduler));
-            var outputPipe = connectionInfo.PipeFactory.Create(GetOutputPipeOptions(connectionInfo.OutputReaderScheduler));
+            var connectionContext = new DefaultConnectionContext(features);
+
+            var transportFeature = connectionContext.Features.Get<IConnectionTransportFeature>();
```

Comment:
When do we ever do that? I could move `DefaultConnectionContext` to Kestrel.Core and put whatever common properties in there though.